services/src/main/java/org/exoplatform/addons/gamification/upgrade/DomainMigrationService.java

```diff
@@ -0,0 +1,214 @@
+                        if (domainDTO != null) {
+                            entitiesToMigrate = true;
+                            rule.setDomainDTO(domainDTO);
+                            ruleService.updateRule(rule);
+                        }
+                    }
+                }
+                long endTime = System.currentTimeMillis();
+                if (entitiesToMigrate) {
+                    LOG.info("=== Migration of Rules is done in " + (endTime - startTime) + " ms");
+                } else {
+                    LOG.info("=== No rules to migrate");
+                }
+                rulesmigrationDone=true;
+            } catch (Exception e) {
+                LOG.error("Error when migration Rules ", e);
+            }
+
+            try {
+                LOG.info("=== Start migration of Badges");
+                startTime = System.currentTimeMillis();
+                entitiesToMigrate = false;
+                List<String> domains = badgeService.getDomainListFromBadges();
+                for (String title : domains) {
+                    DomainDTO domainDTO = domainService.findDomainByTitle(title);
+                    if (domainDTO != null) {
+                        List<BadgeDTO> badges = badgeService.findBadgesByDomain(title);
+                        if (badges != null && badges.size() > 0) {
+                            for (BadgeDTO badge : badges) {
+                                if (badge.getDomainDTO() == null) {
+                                    entitiesToMigrate = true;
+                                    badge.setDomainDTO(domainDTO);
+                                    badgeService.updateBadge(badge);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                List<BadgeDTO> badges = badgeService.getAllBadgesWithNullDomain();
+                if (badges != null && badges.size() > 0) {
+                    entitiesToMigrate = true;
+                    for (BadgeDTO badge : badges) {
+                        DomainDTO domainDTO = domainService.findDomainByTitle(badge.getDomain());
+                        if (domainDTO != null) {
+                            entitiesToMigrate = true;
+                            badge.setDomainDTO(domainDTO);
+                            badgeService.updateBadge(badge);
+                        }
+                    }
+                }
+                long endTime = System.currentTimeMillis();
+                if (entitiesToMigrate) {
+                    LOG.info("=== Migration of Badges is done in " + (endTime - startTime) + " ms");
+                } else {
+                    LOG.info("=== No badges to migrate");
+                }
+                badgesMigrationDone=true;
+            } catch (Exception e) {
+                LOG.error("Error when migration Rules ", e);
```

Comment:
The error is for the badges migration, not rules.

Reply:
ok